src/map.rs

```diff
@@ -947,15 +952,13 @@ where
     fn help_transfer<'g>(
         &'g self,
         table: Shared<'g, Table<K, V>>,
-        next_table: *const Table<K, V>,
+        next_table: Shared<'g, Table<K, V>>,
```

Comment:
Does this even need to take a `next_table` argument now, or can it just load it from the passed `table`?

Reply:
I'm _pretty_ sure that by the time this is called, `table.next_table` has already been set, so `t.next_table()` can be used over `t.get_moved()`.

Reply:
This is in fact always what happens before and is passed as the argument. If you prefer, this could be handled by `help_transfer`

Reply:
Previously I don't think `help_transfer` had any way to learn the address of the next table. Now it does, so I think we should take advantage of that!